### PR TITLE
`prefer-negative-index`: Check `.toSpliced()` and `.with()`

### PR DESCRIPTION
**Pull request by fisker**
Ref #2029

### CODE REVIEW

File: docs/rules/prefer-negative-index.md [hunk 1]
```diff
@@ -7,7 +7,18 @@
...
+- [`String#slice()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/slice)
+- [`Array#slice()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/slice)
+- [`TypedArray#slice()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/TypedArray/slice)
+- [`String#at()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/at)
+- [`Array#at()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/at)
+- [`TypedArray#at()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/TypedArray/at)
+- [`Array#splice()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/splice)
+- [`Array#toSpliced()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/toSpliced)
+- [`Array#with()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/with)
+- [`TypedArray#with()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/TypedArray/with)
```
Comment (fisker) [inline docs/rules/prefer-negative-index.md:21]: Those links are not available yet, but let's assume they will be there soon.

Comment (sindresorhus) [reply docs/rules/prefer-negative-index.md:21]: I cannot merge this with broken links though.

Comment (sindresorhus) [inline docs/rules/prefer-negative-index.md:21]:
```suggestion
- `Array#toSpliced()`
- `Array#with()`
- `TypedArray#with()`

<!--
- [`Array#toSpliced()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/toSpliced)
- [`Array#with()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/with)
- [`TypedArray#with()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/TypedArray/with)
-->
```

File: docs/rules/prefer-negative-index.md [hunk 2]
```diff
@@ -1,4 +1,4 @@
-# Prefer negative index over `.length - index` for `{String,Array,TypedArray}#{slice,at}()` and `Array#splice()`
+# Prefer negative index over `.length - index` for builtin object methods supports relative index
```
Comment (sindresorhus) [inline docs/rules/prefer-negative-index.md:1]:
```suggestion
# Prefer negative index over `.length - index` when possible
```

**Review by sindresorhus — CHANGES_REQUESTED** [review]

File: docs/rules/prefer-negative-index.md (hunk 2 above)
Comment (fisker) [reply docs/rules/prefer-negative-index.md:1]: Much better

**Pull request merged**